core/src/main/java/org/apache/accumulo/core/file/blockfile/impl/SeekableByteArrayInputStream.java

```diff
@@ -78,14 +80,14 @@ public int read(byte[] b, int offset, int length) {
       length = avail;
     }
 
-    System.arraycopy(buffer, cur, b, offset, length);
-    cur += length;
+    System.arraycopy(buffer, cur.get(), b, offset, length);
+    cur.getAndAdd(length);
```

Comment:
This section is another section where the value could change, and it's not thread-safe, between the various non-atomic calls to `cur.get()`. There's one hidden in the `available()` call, and then one in the array copy, then the subsequent advancement of the pointer.

If we want this to be thread-safe, it should probably get the current value and advance the pointer atomically, so that outside threads can't change the pointer during the same call, similar to my previous suggestion.

Reply:
addressed in f251078